### PR TITLE
feat(module/vmseries): bootstrap options validation - instance metadata

### PR DESCRIPTION
**Pull request by FoSix**
## Description

Add validation of bootstrap parameters.

## Motivation and Context

Minimise possibility to pass wrong bootstrap options.

## How Has This Been Tested?

Following values were set to test the validation rule:
test | result
--- | ---
`"type=dhcp-client"` | :-1:
`{}` | :+1:
`{`<br>`hostname           = "vms01"`<br>`panorama-server    = "10.1.2.3"`<br>`plugin-op-commands = "numa-perf-optimize:enable,set-dp-cores:2"`<br>`type               = "dhcp-client"`<br>`}` | :+1:
  

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/vmseries/variables.tf [hunk 1]
```diff
@@ -30,9 +30,32 @@ variable "network_interfaces" {
 }
 
 variable "bootstrap_options" {
-  description = "VM-Series bootstrap options to pass using instance metadata."
+  description = <<-EOF
+  VM-Series bootstrap options to pass using instance metadata.
+
+  Proper syntax is a map where keys are the bootstrap properties.
```
Comment (michalbil) [inline modules/vmseries/variables.tf:36]:
```suggestion
  Proper syntax is a map, where keys are the bootstrap parameters.
```
I'd propose to stick to wording used in [bootstrap options doc](https://docs.paloaltonetworks.com/vm-series/10-2/vm-series-deployment/bootstrap-the-vm-series-firewall/create-the-init-cfgtxt-file/init-cfgtxt-file-components).

File: modules/vmseries/variables.tf [hunk 2]
```diff
@@ -30,9 +30,32 @@ variable "network_interfaces" {
...
+  Example:
+    bootstrap_options = {
+      type            = dhcp-client
+      panorama-server = 1.2.3.4
+    }
+
+  A list of available keys/properties: type, ip-address, default-gateway, netmask, ipv6-address, ipv6-default-gateway, hostname, panorama-server, panorama-server-2, tplname, dgname, dns-primary, dns-secondary, vm-auth-key, op-command-modes, op-cmd-dpdk-pkt-io, plugin-op-commands, dhcp-send-hostname, dhcp-send-client-id, dhcp-accept-server-hostname, dhcp-accept-server-domain, auth-key.
```
Comment (michalbil) [inline modules/vmseries/variables.tf:43]:
```suggestion
  A list of available parameters: type, ip-address, default-gateway, netmask, ipv6-address, ipv6-default-gateway, hostname, panorama-server, panorama-server-2, tplname, dgname, dns-primary, dns-secondary, vm-auth-key, op-command-modes, op-cmd-dpdk-pkt-io, plugin-op-commands, dhcp-send-hostname, dhcp-send-client-id, dhcp-accept-server-hostname, dhcp-accept-server-domain, auth-key.
```

File: modules/vmseries/variables.tf [hunk 3]
```diff
@@ -30,9 +30,32 @@ variable "network_interfaces" {
...
+
+  For more details on bootstrapping see documentation: https://docs.paloaltonetworks.com/vm-series/10-2/vm-series-deployment/bootstrap-the-vm-series-firewall/create-the-init-cfgtxt-file/init-cfgtxt-file-components
```
Comment (michalbil) [inline modules/vmseries/variables.tf:45]:
```suggestion
  For more details on the options please refer to [VM-Series documentation](https://docs.paloaltonetworks.com/vm-series/10-2/vm-series-deployment/bootstrap-the-vm-series-firewall/create-the-init-cfgtxt-file/init-cfgtxt-file-components).
```

**Review by michalbil — APPROVED** [review]
LGTM!

**Review by michalbil — APPROVED** [review]

**Review by migara — APPROVED** [review]

**Pull request merged**